app/api/undo.py

```
@@ -199,33 +185,33 @@ def undo_to_snapshot(
                 raise
 
     raise RuntimeError(f"Failed to undo to snapshot after {max_retries} retries")
 
 
 @router.get("/{session_id}/snapshots")
-def list_session_snapshots(
+async def list_session_snapshots(
     session_id: int,
     current_user: Annotated[User, Depends(get_current_user)],
-    db: Session = Depends(get_db),
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)